internal/server/data/destination.go

```diff
@@ -67,12 +68,89 @@ func UpdateDestination(tx WriteTxn, destination *models.Destination) error {
 	return update(tx, (*destinationsUpdateTable)(destination))
 }
 
-func GetDestination(db GormTxn, selectors ...SelectorFunc) (*models.Destination, error) {
-	return get[models.Destination](db, selectors...)
+type GetDestinationOptions struct {
+	// ByID instructs GetDestination to return the row matching this ID. When
+	// this value is set, all other fields on this strut will be ignored
+	ByID uid.ID
+	// ByUniqueID instructs GetDestination to return the row matching this
+	// uniqueID.
+	ByUniqueID string
 }
 
-func ListDestinations(db GormTxn, p *Pagination, selectors ...SelectorFunc) ([]models.Destination, error) {
-	return list[models.Destination](db, p, selectors...)
+func GetDestination(tx ReadTxn, opts GetDestinationOptions) (*models.Destination, error) {
+	destination := destinationsTable{}
+	query := querybuilder.New("SELECT")
+	query.B(columnsForSelect(destination))
+	query.B("FROM destinations")
+	query.B("WHERE deleted_at is null AND organization_id = ?", tx.OrganizationID())
+
+	switch {
+	case opts.ByID != 0:
+		query.B("AND id = ?", opts.ByID)
+	case opts.ByUniqueID != "":
+		query.B("AND unique_id = ?", opts.ByUniqueID)
+	default:
+		return nil, fmt.Errorf("an ID is required to GetDestination")
+	}
+
+	err := tx.QueryRow(query.String(), query.Args...).Scan(destination.ScanFields()...)
+	if err != nil {
+		return nil, handleReadError(err)
+	}
+	return (*models.Destination)(&destination), nil
+}
+
+type ListDestinationsOptions struct {
+	ByUniqueID string
+	// TODO: name is not indexed
+	ByName string
```

Comment:
I think we'll likely need to restore the unique constraint on destination.name, which will act as the index for these queries.

Reply:
#3392